src/utils/atomWithStorage.ts

```diff
@@ -1,9 +1,13 @@
 import { atom } from 'jotai'
-import type { SetStateAction, WritableAtom } from 'jotai'
+import type { WritableAtom } from 'jotai'
 import { RESET } from './constants'
 
 type Unsubscribe = () => void
 
+type SetStateActionFunc<Value> = (prev: Value) => Value | typeof RESET
+
+type StorageSetStateAction<Value> = Value | SetStateActionFunc<Value>
```

Comment:
How about this?
```suggestion
type SetStateActionWithReset<Value> = Value | typeof RESET | ((prev: Value) => Value | typeof RESET)
```
Just a preference.

Reply:
Yeah, it should be better as we can remove most of `typeof RESET` below.

Reply:
Initially did the function type to help with func typecasting, but since this can't be reused anywhere I will stick with the repo conventions!

1aa8917